### PR TITLE
Feature/action cable implementation

### PR DESCRIPTION
**Pull request by ElAtrash**

### CODE REVIEW

File: app/controllers/messages_controller.rb [hunk 1]
```diff
@@ -3,13 +3,26 @@ class MessagesController < ApplicationController
 
   def index
     messages = @conversation.messages
+    messages.where(status: 0).update(status: 1)
```
Comment (vkalach) [inline app/controllers/messages_controller.rb:6]: messages.read

enum features

read - all messages with status read
read! - move message to status read
read? - check if the status is read

File: app/javascript/channels/notifications_channel.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import consumer from "./consumer";
+import "bootstrap";
```
Comment (vkalach) [inline app/javascript/channels/notifications_channel.js:2]: remove it

File: app/controllers/messages_controller.rb [hunk 3]
```diff
@@ -3,13 +3,26 @@ class MessagesController < ApplicationController
...
     message = @conversation.messages.new
     render locals: { conversation: @conversation, messages: messages, message: message }
   end
 
   def create
     message = @conversation.messages.create!(message_params)
-    redirect_to conversation_messages_path(@conversation) if message.save
+
+    if message.save
+      html = render(
```
Comment (mrusakovich) [inline app/controllers/messages_controller.rb:15]: one line, without braces

File: app/controllers/messages_controller.rb [hunk 4]
```diff
@@ -3,13 +3,26 @@ class MessagesController < ApplicationController
...
+        partial: 'messages/message',
+        locals: { message: message }
+      )
+      if @conversation.sender == current_user
+        Notification.create(recipient: @conversation.recipient, user: current_user, action: 'messaged', notifiable: @conversation.recipient)
+      else
+        Notification.create(recipient: @conversation.sender, user: current_user, action: 'messaged', notifiable: @conversation.sender)
+      end
+      ActionCable.server.broadcast 'conversation_channel', { html: html }
```
Comment (mrusakovich) [inline app/controllers/messages_controller.rb:24]: last argument without curly braces, if it's a Hash

File: app/controllers/messages_controller.rb [hunk 5]
```diff
@@ -3,13 +3,26 @@ class MessagesController < ApplicationController
...
   def create
     message = @conversation.messages.create!(message_params)
-    redirect_to conversation_messages_path(@conversation) if message.save
+
+    if message.save
+      html = render(
+        partial: 'messages/message',
+        locals: { message: message }
+      )
+      if @conversation.sender == current_user
```
Comment (mrusakovich) [inline app/controllers/messages_controller.rb:19]: 2nd level if - hard to read, slower to perform for processor https://en.wikipedia.org/wiki/Branch_predictor

File: app/jobs/notification_relay_job.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class NotificationRelayJob < ApplicationJob
+  queue_as :default
+
+  def perform(notification)
+    # ActionCable.server.broadcast "notifications:#{notification.recipient_id}", message: 'You recieved a new message'
+    html = ApplicationController.render partial: "notifications/#{notification.notifiable_type.underscore.pluralize}/#{notification.action}", locals: { notification: notification }, formats: [:html]
+    ActionCable.server.broadcast "notifications:#{notification.recipient_id}", { html: html }
```
Comment (mrusakovich) [inline app/jobs/notification_relay_job.rb:7]: redundant curly braces for the last argument

**Pull request merged**